### DISCUSSION TITLE
Add `max_token` parameter support for API models

### DISCUSSION THREAD

**Discussion by CookSleep** [open-webui/open-webui#5448] [General]
**Is your feature request related to a problem? Please describe.**
Continuing the issue from #4983, I only realized today that `Context Length` is completely ineffective for API models and only works for Ollama models...
We need the `max_token` parameter to manage the context to ensure API costs are consumed within the expected range

**Describe the solution you'd like**
Add `max_token` parameter support for API models

**Describe alternatives you've considered**
Functions like `Context Clip Filter` cannot replace this functionality in API parameters, and even if some other functions implement the feature to trim the context to only the specified number of tokens, many people won't visit the Open WebUI community to load it.

**Additional context**
I really think that many features of Open WebUI **should not** rely on functions or plugins to implement. They are really basic (such as `max_token` and `Max Message Count in Context`, etc.), and not everyone has the patience or can discover that the community has the functions they need... Please seriously consider this point.

**Comment by tjbck** [comment]
`max_token` has been supported from the beginning, please check the chat payload and let me know if its not being included in the request somehow. https://github.com/open-webui/open-webui/blob/b64c9d966a6498d4f2ffe12ab1498fab298afb37/src/lib/components/chat/Chat.svelte#L1208

**Reply by CookSleep** [reply]
How should it be set? I haven't found any parameters that can affect it... Neither `Context Length` nor `Max Tokens (num_predict)` have any effect on the `max_token` of the API model... The model still remembers content far beyond the set limit, and the token consumption in the API logs proves this

**Comment by gamdud** [comment]
https://github.com/open-webui/open-webui/discussions/5005#discussioncomment-12016415

I'm having the same issue as the above and OP. The max_tokens parameter doesn't seem to be doing anything. Default logs of the container didn't reveal payload information. LLM confirmed to have a working max_tokens parameter.

The payload that gets sent from the browser to the open webui backend does have the max_tokens inside the "params" subelement. I don't know what the open webui backend does with it after. 

Edit. Bugfix merged three days ago to dev. https://github.com/open-webui/open-webui/pull/9082

**Comment by Urammar** [comment]
Re-opening this if possible, max_tokens does absolutely nothing. Possibly related to thinking tokens?
